packages/bcp/src/cosmwasmconnection.ts

```diff
@@ -172,26 +171,31 @@ export class CosmWasmConnection implements BlockchainConnection {
           const response = JSON.parse(
             await this.restClient.queryContractSmart(erc20.contractAddress, queryMsg),
           );
+          const normalizedBalance = new BN(response.balance).toString();
           return {
             fractionalDigits: erc20.fractionalDigits,
-            quantity: response.balance,
+            quantity: normalizedBalance,
             tokenTicker: erc20.ticker as TokenTicker,
           };
         },
       ),
     );
+    const nonZeroErc20Amounts = erc20Amounts.filter(amount => amount.quantity !== "0");
```

Comment:
This is both cool, and expensive if we have more than say 5-10 erc20 defined. (Have you seen the list on myetherwallet... it is like 100+). Anyway, if the problem we have is too many tokens on the chain, then we are doing good :joy:

Reply:
The bigger problem I see there is the lack of a proper token identifier. Optimizing working code is relatively easy compared to writing a cross chain spec.